app/models/account.rb

```diff
@@ -74,14 +75,22 @@ def switch
     reset!
   end
 
+  # does not use confirm_endpoints! because we can still nil-ify settings
   def reset!
-    solr_endpoint.reset! if solr_endpoint
-    fcrepo_endpoint.reset! if fcrepo_endpoint
-    redis_endpoint.reset! if redis_endpoint
+    SolrEndpoint.reset!
+    FcrepoEndpoint.reset!
+    RedisEndpoint.reset!
   end
 
   private
 
+    # @raise [RuntimeError] if missing any endpoint
+    def confirm_endpoints!
+      raise "Account #{cname} is missing solr_endpoint, cannot switch!" unless solr_endpoint
```

Comment:
@cbeer what is the utility of a novel exception class that should never be caught?  If you want one, might'nt you prefer three?